File: CSharp/AboutBoxForm.cs
using System;
using System.ComponentModel;
using System.Diagnostics;
using System.Reflection;
using System.Windows.Forms;

using Vintasoft.Barcode;

namespace BarcodeDemo
{
    /// <summary>
    /// A form that shows information about application.
    /// </summary>
    public partial class AboutBoxForm : Form
    {

        #region Constructors

        /// <summary>
        /// Initializes a new instance of the <see cref="AboutBoxForm"/> class.
        /// </summary>
        public AboutBoxForm()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="AboutBoxForm"/> class.
        /// </summary>
        /// <param name="productPrefix">The product prefix.</param>
        public AboutBoxForm(string productPrefix)
            : this()
        {
            nameLabel.Text = string.Format(nameLabel.Text, AssemblyTitle, AssemblyShortVersion);
            imagingSDKVersionLabel.Text = string.Format(imagingSDKVersionLabel.Text, AssemblyVersion);
            productLinkLabel.Text = string.Format(productLinkLabel.Text, productPrefix);
            productFAQLinkLabel.Text = string.Format(productFAQLinkLabel.Text, productPrefix);
        }

        #endregion



        #region Properties

        /// <summary>
        /// Gets the assembly title.
        /// </summary>
        [Browsable(false)]
        public string AssemblyTitle
        {
            get
            {
                object[] attributes = Assembly.GetEntryAssembly().GetCustomAttributes(typeof(AssemblyTitleAttribute), false);
                if (attributes.Length > 0)
                {
                    AssemblyTitleAttribute titleAttribute = (AssemblyTitleAttribute)attributes[0];
                    if (titleAttribute.Title != "")
                    {
                        return titleAttribute.Title;
                    }
                }
                return System.IO.Path.GetFileNameWithoutExtension(Assembly.GetExecutingAssembly().CodeBase);
            }
        }

        /// <summary>
        /// Gets the assembly version.
        /// </summary>
        [Browsable(false)]
        public string AssemblyVersion
        {
            get
            {
                return Assembly.GetAssembly(typeof(IBarcodeInfo)).GetName().Version.ToString();
            }
        }

        /// <summary>
        /// Gets the assembly short version.
        /// </summary>
        [Browsable(false)]
        public string AssemblyShortVersion
        {
            get
            {
                Version ver = Assembly.GetAssembly(typeof(IBarcodeInfo)).GetName().Version;
                return string.Format("{0}.{1}", ver.Major, ver.Minor);
            }
        }

        #endregion



        #region Methods

        /// <summary>
        /// Raises the System.Windows.Forms.Form.Load event.
        /// </summary>
        protected override void OnLoad(EventArgs e)
        {
            Text = "About...";
            base.OnLoad(e);
        }


        private void okButton_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void linkLabel_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            ProcessStartInfo processInfo = new ProcessStartInfo(string.Format("https://{0}", ((LinkLabel)sender).Text));
            processInfo.UseShellExecute = true;
            Process.Start(processInfo);
        }

        private void vintasoftLogoPictureBox_Click(object sender, EventArgs e)
        {
            ProcessStartInfo processInfo = new ProcessStartInfo("https://www.vintasoft.com");
            processInfo.UseShellExecute = true;
            Process.Start(processInfo);
        }

        #endregion

    }
}

File: VB.NET/AboutBoxForm.vb
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Reflection
Imports System.Windows.Forms
Imports Vintasoft.Barcode

Public Partial Class AboutBoxForm
	Inherits Form

	#Region "Constructors"

	Public Sub New()
		InitializeComponent()
	End Sub

	Public Sub New(productPrefix As String)
		Me.New()
		nameLabel.Text = String.Format(nameLabel.Text, AssemblyTitle, AssemblyShortVersion)
		imagingSDKVersionLabel.Text = String.Format(imagingSDKVersionLabel.Text, AssemblyVersion)
		productLinkLabel.Text = String.Format(productLinkLabel.Text, productPrefix)
		productFAQLinkLabel.Text = String.Format(productFAQLinkLabel.Text, productPrefix)
	End Sub

	#End Region



	#Region "Properties"

	<Browsable(False)> _
	Public ReadOnly Property AssemblyTitle() As String
		Get

			Dim attributes As Object() = Assembly.GetEntryAssembly().GetCustomAttributes(GetType(AssemblyTitleAttribute), False)
			If attributes.Length > 0 Then
				Dim titleAttribute As AssemblyTitleAttribute = DirectCast(attributes(0), AssemblyTitleAttribute)
				If titleAttribute.Title <> "" Then
					Return titleAttribute.Title
				End If
			End If
			Return System.IO.Path.GetFileNameWithoutExtension(Assembly.GetExecutingAssembly().CodeBase)
		End Get
	End Property

	<Browsable(False)> _
	Public ReadOnly Property AssemblyVersion() As String
		Get
			Return Assembly.GetAssembly(GetType(IBarcodeInfo)).GetName().Version.ToString()
		End Get
	End Property

	<Browsable(False)> _
	Public ReadOnly Property AssemblyShortVersion() As String
		Get
			Dim ver As Version = Assembly.GetAssembly(GetType(IBarcodeInfo)).GetName().Version
			Return String.Format("{0}.{1}", ver.Major, ver.Minor)
		End Get
	End Property

	#End Region



	#Region "Methods"

	Protected Overrides Sub OnLoad(e As EventArgs)
		Text = "About..."
		MyBase.OnLoad(e)
	End Sub

    Private Sub okButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles okButton.Click
        Close()
    End Sub

    Private Sub linkLabel_LinkClicked(ByVal sender As Object, ByVal e As LinkLabelLinkClickedEventArgs) Handles registerLinkLabel.LinkClicked, productLinkLabel.LinkClicked, productFAQLinkLabel.LinkClicked, forumsLinkLabel.LinkClicked
        Process.Start(String.Format("http://{0}", DirectCast(sender, LinkLabel).Text))
    End Sub

    Private Sub vintasoftLogoPictureBox_Click(ByVal sender As Object, ByVal e As EventArgs) Handles vintasoftLogoPictureBox.Click
        Process.Start("http://www.vintasoft.com")
    End Sub

	#End Region

End Class
